trolo/export/exporter.py

```diff
@@ -0,0 +1,154 @@
+            else:
+                LOGGER.warning("Config not found in checkpoint, inferring from model name")
+                config = infer_model_config_path(model)
+                self.config = self.load_config(config)
+        else:
+            # Convert config to path if it's a name
+            if isinstance(config, str) and not Path(config).exists():
+                config = get_model_config_path(config)
+            self.config = self.load_config(config)
+
+        super().__init__(model, device)
+
+    def load_config(self, config_path: str) -> Dict:
+        """Load config from YAML"""
+        LOGGER.info(f"Loading config from {config_path}")
+        cfg = YAMLConfig(config_path)
+        return cfg
+    
+    def load_model(self, model: str) -> torch.nn.Module:
+        """Load detection model using config"""
+        # Load checkpoint
+        checkpoint = torch.load(model, map_location="cpu", weights_only=False)
+
+        if "HGNetv2" in self.config.yaml_cfg:
+            self.config.yaml_cfg["HGNetv2"]["pretrained"] = False
+
+        # Load model state
+        if "ema" in checkpoint:
+            state = checkpoint["ema"]["module"]
+        else:
+            state = checkpoint["model"]
+
+        # Load state into config.model
+        self.config.model.load_state_dict(state)
+
+        # Create deployment model wrapper
+        model = self.config.model.deploy()
+
+        return model
+
+    def export(
+        self, 
+        export_format ,
+        input_size ,
+        input_data , 
+        input_names  , 
+        output_names ,
+        dynamic_axes,
+        batch_size : int =  1,
+        opset_version : int = 16,
+        simplify : bool = False,
+    ):
+        # check the model format
+        if export_format is None:
+            raise ValueError(f"Export format is missing!")
+        
+        #  check for export format
+        accepted = self._filter_format(export_format)
+
+        if export_format.lower().strip() == "onnx":
+            self._export2onnx(
+                input_size=input_size, 
+                input_data = input_data, 
+                input_names =  input_names, 
+                output_names = output_names,
+                dynamic_axes= dynamic_axes,
+                batch_size = batch_size, 
+                simplify = simplify, 
+                opset_version = opset_version
+            )
+
+
+    def _export2onnx(
+        self,
+        input_size : Union[int, List, Tuple, torch.Tensor] = None,
+        input_data : Union[int, List, Tuple, torch.Tensor] = None, 
+        input_names : Optional[list] =  None, 
+        output_names : Optional[list] =  None,
+        dynamic_axes : Optional [dict] =  None,
+        batch_size : int =  1,
+        opset_version : int = 16,
+        simplify : bool = False,
+        export_format : str = "onnx"
+    ) -> None: 
+        input_size  = format_size(input_size)
+        input_data  = format_input(input_data, batch_size=batch_size)
```

Comment:
Use random tensor and remove `format_input` fn

Reply:
can we not create separate function for `input_size` and make changes here only? Otherwise for other format, the function will grow and it will become hard to maintain

Reply:
updated